tests/src/Kernel/RulesMailTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\Tests\rules\Kernel;
+
+/**
+ * Performs tests on the pluggable mailing framework.
+ *
+ * @group Mail
+ */
+class RulesMailTest extends RulesDrupalTestBase {
+
+  /**
+   * The action manager used to instantiate the action plugin.
+   *
+   * @var \Drupal\rules\Core\RulesActionManagerInterface
+   */
+  protected $actionManager;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['rules'];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    // Use the state system collector mail backend.
+    $this->container->get('config.factory')->getEditable('system.mail')
+      ->set('interface.default', 'test_mail_collector')
+      ->save();
+
+    // Reset the state variable that holds sent messages.
+    $this->container->get('state')->set('system.test_mail_collector', []);
+
+    $this->actionManager = $this->container->get('plugin.manager.rules_action');
+  }
+
+  /**
+   * Checks the From: and Reply-to: headers.
+   */
+  public function testSubjectAndBody() {
+    // Create action to send email.
+    $action = $this->actionManager->createInstance('rules_send_email');
+
+    $params = [
+      'to' => ['mail@example.com'],
+      'subject' => 'subject',
+      'message' => 'hello',
+    ];
```

Comment:
Now that $params variable is not needed anymore. Just use the literals when setting the context values and when asserting the outcome.